### PR TITLE
1249 API to get store preferences 

### LINKED ISSUE #1249: API to get store preferences
## Is your feature request related to a problem? Please describe 👀
Related to #1169. Will need store preferences to load customised UIs in programs.

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1249 

# 👩🏻‍💻 What does this PR do? 
Endpoint to query for store preferences

# 🧪 How has/should this change been tested? 
Call endpoint to see all store preferences

- Will need to update programs with develop once this is merged

### CODE REVIEW

File: server/service/src/auth.rs [hunk 1]
```diff
@@ -244,6 +245,11 @@ fn all_permissions() -> HashMap<Resource, PermissionDSL> {
         Resource::QueryInventoryAdjustmentReasons,
         PermissionDSL::NoPermissionRequired,
     );
+    map.insert(
+        Resource::QueryStorePreferences,
+        PermissionDSL::NoPermissionRequired,
```
Comment (clemens-msupply) [inline server/service/src/auth.rs:250]: Don't you need store permisisons? `PermissionDSL::HasStoreAccess`

Comment (roxy-dao) [reply server/service/src/auth.rs]: [716b173](https://github.com/openmsupply/open-msupply/pull/1250/commits/716b173bf1fffdcd625abc7f7724ffa28c016f10)

Comment (roxy-dao) [reply server/service/src/auth.rs]: [ff19359](https://github.com/openmsupply/open-msupply/pull/1250/commits/ff193598e53cbe5e271a52069ee47c0703a463ba)

**Comment by mark-prins** [comment]
should the PR title and branch be 'preferences'?

**Review by mark-prins — APPROVED** [review]
👍 changes look fine to me

**Comment by roxy-dao** [comment]
> should the PR title and branch be 'preferences'?
> 👍 changes look fine to me

:facepalm: I knew that... 😭 
- Setting up the hooks in FE, so have changed this back to draft (:

**Pull request merged**